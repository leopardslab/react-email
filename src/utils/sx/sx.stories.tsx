import { ComponentStory, ComponentMeta } from '@storybook/react';

import { sx } from './sx';

const styles = {
  primary: {
    width: 'fit-content',
    backgroundColor: 'yellow',
    padding: '50px',
  },
  secondary: {
    padding: '10px',
    color: 'red',
  },
};

const Email = (): JSX.Element => {
  const style = sx(styles.primary, styles.secondary);
  return (
    <div style={style}>
      <div>
        <h2>Hello World</h2>
      </div>
    </div>
  );
};

export default {
  component: Email,
} as ComponentMeta<typeof Email>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Email> = () => <Email />;

export const Default = Template.bind({});
