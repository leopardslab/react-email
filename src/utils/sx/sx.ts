import { CSSProperties } from 'react';

type SXProps = CSSProperties | false | undefined;

export const sx = (...args: SXProps[]): CSSProperties => {
  const initialValue: CSSProperties = {};

  const result = args.reduce((prev: CSSProperties, curr: SXProps) => {
    if (!curr) {
      return prev;
    }
    return { ...prev, ...curr };
  }, initialValue);

  return result;
};
