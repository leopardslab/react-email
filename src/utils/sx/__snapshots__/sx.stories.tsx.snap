// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/sx Default smoke-test 1`] = `
<div style="width: fit-content; background-color: yellow; padding: 10px; color: red;">
  <div>
    <h2>
      Hello World
    </h2>
  </div>
</div>
`;
